### PR TITLE
Finish Auth Signup Flow

### PR DESCRIPTION
**Pull request by meganniu**
Changes:
+ If the user closes SignUpModal, they are redirected to the donor homepage
+ If the user successfully creates an account, they are redirected to the ConfirmationModal
+ If the user closes ConfirmationModal, they are redirected to the donor homepage

https://user-images.githubusercontent.com/20423590/113802082-1151ed80-9728-11eb-85d4-e2fd017c10a5.mov

### CODE REVIEW

**Review by chamod-gamage — APPROVED** [review]
Your changes look good to me but I would wait until #104 gets merged (due to modal styling changes and the potential for merge conflicts) before merging this PR.

**Review by guy-ro — APPROVED** [review]
LGTM

**Pull request merged**
